### ISSUE TITLE
Queue name validation

### ISSUE THREAD

**Issue by pzielinski86** [HangfireIO/Hangfire#550]
We added a task with a queue name containing a dash ('-'). We then tried to trigger it on the dashboard but it just hang on _"Triggering..."_ and it did nothing.

After investigating, we found that [there is a regular expression](https://github.com/HangfireIO/Hangfire/blob/2126550aaf52870122548474b89b1630dbb7d2c8/src/Hangfire.Core/States/EnqueuedState.cs#L226), permitting only letters, digits and underscore.

1. Is this is a bug that it accepts a task with a dash (in the queue name) but can't trigger it?
2. If it's not a bug, then maybe there should be an error on the dashboard explaining why.

Thanks

**Comment by pzielinski86** [comment]
We have just discovered that it doesn't even accept upper case. For example "Test" is invalid name (it should be "test").

**Comment by AbubakerB** [comment]
Yes, i have the same issue!
It seems odd that it accepts it but then can't run it

**Comment by IanYates** [comment]
See the docs: http://docs.hangfire.io/en/latest/background-processing/configuring-queues.html

```
The Queue name argument must consist of lowercase letters, digits and underscore characters only.
```

So I guess, at the least, an exception should be thrown if you supply a name that doesn't conform.

**Comment by odinserj** [comment]
Closing – was fixed by PR #561 in 1.5.7.
